File: Data Structure & Algorithms/DSA_Level-1/Basics Of Programming/Function And Arrays/Any Base Addition/AnyBaseAddition_Solution.js

function addition(b,n1,n2){
    let c=0;
    let pow=0;
    let res=0;
    while(c>0 || n1>0 || n2>0){
        let temp1=n1%10, temp2=n2%10;
        res+=((c+temp1+temp2)%b)*Math.pow(10,pow);
        c=parseInt((temp1+temp2+c)/b);
        pow++;
        n1=parseInt(n1/10);
        n2=parseInt(n2/10);
    }

    return res;
}

console.log(addition(8,777,1));

File: Data Structure & Algorithms/DSA_Level-1/Basics Of Programming/Function And Arrays/Any Base Addition/AnyBaseAddition_Solution.java
import java.util.*;

public class Main {

  public static void main(String[] args) {
    Scanner scn = new Scanner(System.in);
    int b = scn.nextInt();
    int n1 = scn.nextInt();
    int n2 = scn.nextInt();

    int d = getSum(b, n1, n2);
    System.out.println(d);
  }

  public static int getSum(int b, int n1, int n2) {
    int c = 0;
    int pow = 0;
    int res = 0;
    while (c > 0 || n1 > 0 || n2 > 0) {
      int temp1 = n1 % 10, temp2 = n2 % 10;
      res += ((c + temp1 + temp2) % b) * Math.pow(10, pow);
      c = (int)(temp1 + temp2 + c) / b;
      pow++;
      n1 = (int)n1 / 10;
      n2 = (int)n2 / 10;
    }

    return res;
  }
}